src/WrappedMToken.sol

```diff
@@ -81,15 +81,37 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
 
     /// @inheritdoc IWrappedMToken
     function wrap(address recipient_, uint256 amount_) external {
+        _wrap(msg.sender, recipient_, UIntMath.safe240(amount_));
+    }
+
+    /// @inheritdoc IWrappedMToken
+    function wrap(address recipient_) external {
+        _wrap(msg.sender, recipient_, UIntMath.safe240(IMTokenLike(mToken).balanceOf(msg.sender)));
+    }
+
+    function _wrap(address account_, address recipient_, uint240 amount_) internal {
         // NOTE: The behavior of `IMTokenLike.transferFrom` is known, so its return can be ignored.
-        IMTokenLike(mToken).transferFrom(msg.sender, address(this), amount_);
+        IMTokenLike(mToken).transferFrom(account_, address(this), amount_);
 
-        _mint(recipient_, UIntMath.safe240(amount_));
+        _mint(recipient_, amount_);
     }
 
     /// @inheritdoc IWrappedMToken
     function unwrap(address recipient_, uint256 amount_) external {
-        _burn(msg.sender, UIntMath.safe240(amount_));
+        _unwrap(msg.sender, recipient_, UIntMath.safe240(amount_));
```

Comment:
We should claim here too. Cause you could call `unwrap` with `amount_` equal to `balanceOf(account) + accruedYieldOf(account)` and it would revert.

Reply:
`_unwrap` already claims, but it was done specifically in this one up front in order to arrive at a final balance. I update the PR to used `accruedYieldOf` instead, so it looks more consistent.